src/parcsr_mv/par_csr_matvec.c

```diff
@@ -117,167 +116,137 @@ hypre_ParCSRMatrixMatvecOutOfPlace( HYPRE_Complex       alpha,
       comm_pkg = hypre_ParCSRMatrixCommPkg(A);
    }
 
+   /* Update send_map_starts, send_map_elmts, and recv_vec_starts when doing
+      sparse matrix/multivector product  */
+   hypre_ParCSRCommPkgUpdateVecStarts(comm_pkg, x);
```

Comment:
It looks like this is one of the main places where the multivector stuff comes in right now, right?  Where else?

Reply:
Yes, this is the main place. For `ParCSRMatrixMatvecT`, this one is also important:
https://github.com/hypre-space/hypre/blob/6a47f0872071dc6e907c8e5f8bf50f2a1b2ecbbc/src/parcsr_mv/par_csr_matop_device.c#L918-L952

Note how `matrix_E` depends on `num_components` now... `matrix_E` is used for unpacking data received from other processors at https://github.com/hypre-space/hypre/blob/6a47f0872071dc6e907c8e5f8bf50f2a1b2ecbbc/src/parcsr_mv/par_csr_matvec.c#L623

Reply:
Thanks for these pointers and the additional info!  This will help the other reviewers as well I think.